### ISSUE TITLE
[Feature] improve search for games with special characters in the title but were stripped from the file name

### ISSUE THREAD

**Issue by Betonhaus** [rommapp/romm#1204]
an example would be games such as Q*bert or Fate: Gates of Dawn which may have file names like Q-Bert.zip or Fate - Gates of Dawn.zip. right now romm can't find them so you have to manually search with special characters removed. Ideally RomM should be able to replace all UNIX special characters in the file name (such as '-') with whitespace in the search, and recognize that titles that have UNIX special characters will perfectly match files that don't have those special characters.

if the automatic identification is too risky, then at least have manual match do a initial search that strips unix special characters and the extension

**Comment by barto28** [comment]
Yep I agree with this. This could make the scan feature way more efficient. A lot of games includes a special character in their title that immediately exclude them from being parsed in the library !

This was labeled as a feature request but isn't more like a bug in the search function ?

Thanks !

**Comment by DDriggs00** [comment]
Most roms are named using the No-Intro Naming convention, which specifies that the following Low ASCII characters are allowed:
`a-z A-Z 0-9 SPACE $ ! # % ' ( ) + , - . ; = @ [ ] ^ _ { } ~ `

**Comment by Betonhaus** [comment]
Not Q*Bert, Fate/Extra, or Blind Fate: Edo no Yami for example 

There are a LOT that use *, :, or / which are unsafe for file names

**Comment by DDriggs00** [comment]
That's what I was referring to. Any characters not in that list above (such as `*`, `:`, or `é`) will be replaced with either `-` or the corresponding letter (`e`) by anything that follows the most widely used naming convention, so will likely not be properly matched.

**Comment by gantoine** [comment]
This has been improved in [v4.0.0](https://github.com/rommapp/romm/releases/tag/4.0.0)
